docs/PowerAuth-Server-0.22.0.md

```diff
@@ -51,6 +51,8 @@ Make sure you upgrade the web container to a version which supports Java 11 befo
 Following DB changes occurred between version 0.21.0 and 0.22.0:
 - Table `pa_activation_history` - added column `external_user_id`
 - Added tables, sequences and indexes for storage of Recovery Codes, Recovery PUKs and Recovery Configuration
+- Table `pa_application` - added not null constraint to column `name` (optional)
```

Comment:
@romanstrobl Should we push the clients to apply those new constraints or there can be a case where they have already stored data which will block this (null name values, or not unique values). Then we should obviously prepare a fix and set default `''` for name?

Reply:
It is possible that clients have a null name in database (it occured in one case already). I do not think a default '' is a good name. I think it would be better to do a migration, something like: ```UPDATE pa_application SET application_name = "application_" + application_id WHERE application_name IS NULL```

Reply:
Brilliant, for sure that will be a better fix!

But still if there are already two same names now e.g. `MOB_APP` we will not succeed with the unique constraint. It will require manual DB update during the migration, is this acceptable? I could add a note about that.

Reply:
Yes, the UPDATE query would be used during migration right before activating the `NOT NULL` contstraint, that was my point.

Reply:
Ah I see, you are concerned about the UNIQUE index. We can prepare some check to avoid this case during migration, it should not happen.

Reply:
Creation of the unique index will be the check itself but we should ensure the migration script was done (i suppose it is a manual script not ensured by the application start). Otherwise I would rather keep the `findFirstByName` query.